src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducerImpl().get();
+        } else if (type == Excavator.class) {
+            machines = new ExcavatorProducerImpl().get();
+        } else if (type == Truck.class) {
+            machines = new TruckProducerImpl().get();
+        } else {
+            return List.of();
+        }
+
+        return (List<Machine>) machines;
```

Comment:
You can use `switch` instead of this complicated if else, also to make code more readable you should change `List.of()` to `Collections.emptyList()`